### DISCUSSION TITLE
Javascript Compatibility

### DISCUSSION THREAD

**Discussion by unfirthman** [utilyre/barbecue.nvim#65] [Q&A] [answered]
Thank you for this awesome plug-in!

Barbecue works well with Lua and Python files. I am having difficulty getting it to work for Javascript files.

I have eslint and tslint installed via Mason.nvim and nvim.lspconfig. Neovim is updated to 0.8.3. 

Please let me know if I can provide more info.

Thank you.

**Comment by utilyre** [comment]
Hello! You need the `typescript-language-server` (`tsserver`) installed.

**Reply by unfirthman** [reply]
Thank you for your response!

I have installed `typescript-language-server` successfully.

After additional troubleshooting, I've noticed an error message related to `nvim-navic` :

`nvim-navic: Failed to attach to tsserver for current buffer. Already attached to tsserver.`

So it appears to be a problem with my `nvim-navic` config!  Odd that it occurs only with JS associated lsp. Also, when I open a new `.js` file, the error message will alternatively be an issue with `vtsls` or `tsserver`.

I'll be digging into my config to get to the bottom of this! There must be some redundancy I am missing. Thanks for the help!

**Reply by utilyre** [reply]
> I'll be digging into my config to get to the bottom of this! There must be some redundancy I am missing.

The error message indicates that you're trying to call `require'nvim-navic'.attach` manually somewhere in your config. This error happens when barbecue tries to attach navic automatically. See `attach_navic` option in barbecue config.

**Reply by unfirthman** [reply]
I've been exploring `attach_navic` by switching between `attach_navic = true,` and `attach_navic = false,`.

This is indeed toggling whether `nvic` is attached: when set to true, Barbecue works with Lua files, and the same error message appears for JS files as stated before. When set to false, Barbecue is toggled off with Lua files, and there is no error message for JS files.

I've also tried commenting out parts of my LSP config that may conflict. For example I've commented this out:
```

lsp.on_attach(function(client, bufnr)
  local opts = {buffer = bufnr, remap = false}

  if client.name == "eslint" then
      vim.cmd.LspStop('eslint')
      return
  end

  vim.keymap.set("n", "gd", vim.lsp.buf.definition, opts)
  vim.keymap.set("n", "K", vim.lsp.buf.hover, opts)
  vim.keymap.set("n", "<leader>vws", vim.lsp.buf.workspace_symbol, opts)
  vim.keymap.set("n", "<leader>vd", vim.diagnostic.open_float, opts)
  vim.keymap.set("n", "[d", vim.diagnostic.goto_next, opts)
  vim.keymap.set("n", "]d", vim.diagnostic.goto_prev, opts)
  vim.keymap.set("n", "<leader>vca", vim.lsp.buf.code_action, opts)
  vim.keymap.set("n", "<leader>vrr", vim.lsp.buf.references, opts)
  vim.keymap.set("n", "<leader>vrn", vim.lsp.buf.rename, opts)
  vim.keymap.set("i", "<C-h>", vim.lsp.buf.signature_help, opts)
end)
```
My issue is only occurring with `tsserver` and `vtsls` while `css-lsp` and  `lua-language-server` are not affected. I've been picking through my config for anything that could be attaching to a JS lsp server and the code above was the closest I can find so far.

Thanks again for any thoughts you can provide. I'm very intent on figuring this out!

**Reply by utilyre** [reply]
You probably need to open an issue over on nvim-navic repo. Feel free to link to this discussion there!

**Reply by unfirthman** [reply]
Thank you! I will!

**Comment by unfirthman** [comment, marked as answer]
## Looks like this has been resolved!

First, commenting out the code referenced above was helpful. I could possibly have those remaps by wrapping the `nvim-navic` 's `attach` function as mentioned in the `nvim-navic` documentation, but I am keeping it simple for now.

Second, following along with a recent issue on the `nvim-navic` repo:

[ How to handle buffer which has multiple LSP clients available? #89 ](https://github.com/SmiteshP/nvim-navic/issues/89#issue-1579829535)

 I removed all JS lsp's I had under `lsp.ensure_installed` except for `tsserver`, `eslint`, `lua_ls`, and `rust_analyzer`.

This meant removing `vtsls` and `quick_lint_js` from `lsp.ensure_installed`. I also `:MasonUninstall vtsls` to wipe it completely.

Now I have JS compatibility! 

Thanks again for all your information and help. I am happy to add this to your documentation if this is something that needs documenting.
